### PR TITLE
Major reorg of repo

### PR DESCRIPTION
**Pull request by tiffanychu90**
* Add more documentation for scripts/notebooks
* Add more items to catalog
* Update notebooks and scripts to pull from catalog
* Rename notebooks to reflect groupings of work
* Add data-workflow-notes to explain data catalog stuff

### CODE REVIEW

File: catalogs/catalog.yml [hunk 1]
```diff
@@ -68,17 +83,43 @@ sources:
...
+    args:
+      urlpath: s3://city-planning-entitlements/gis/intermediate/toc_eligible_parcels_with_entitlements.geojson
+  # Census
+  # NEED TO FIX THIS BECAUSE IT'S GEOPARQUET, needs to be downloaded first
+  census_analysis_table:
+    driver: parquet
+    description: Census data reshaped to tract-level using 2018 ACS data, in notebooks/B1-census-tract-stats.ipynb.
+    args:
+      urlpath: s3://city-planning-entitlements/data/final/census_analysis_table.parquet
+      engine: pyarrow
```
Comment (tiffanychu90) [inline catalogs/catalog.yml]: @ian-r-rose Are we able to upload geoparquet to catalog? If not, I can make this a regular parquet, and we just always merge in the geometry using census_tracts.

Comment (tiffanychu90) [reply catalogs/catalog.yml]: Using parquet, so we still have to merge in `census_tracts` from the catalog to get the geometry. Would rather have census_analysis_table in the catalog, as opposed to always downloading from S3, then reading it in.

Comment (ian-r-rose) [reply catalogs/catalog.yml]: I think that you are right that geopandas doesn't support reading parquet directly from s3 yet. I'd have to upstream that fix. For the time being, either workaround should be okay, I think

**Comment by ian-r-rose** [comment]
Thanks for the cleanup @tiffanychu90, this looks great.

File: catalogs/catalog.yml (hunk 1 above)
Comment (tiffanychu90) [reply catalogs/catalog.yml]: For the time being, I'm putting read-able s3 files in the catalog.

**Pull request merged**